phalcon/di/service/serviceregistry.zep

```diff
@@ -0,0 +1,115 @@
+namespace Phalcon\Di\Service;
+
+use Phalcon\DiInterface;
+use Phalcon\Di\Exception;
+use Phalcon\Di\ServiceInterface;
+use Phalcon\Config\Adapter\Yaml;
+
+/**
+ * Phalcon\Di\Service\ServiceRegistry
+ *
+ **/
+class ServiceRegistry
+{
+
+	/**
+	 *
+	 */
+	protected di;
+
+	/**
+	 *
+	 */
+	public function __construct(<DiInterface> di)
+	{
+		let this->di = di;
+	}
+
+	/**
+	 *
+	 */
+	public function registerDir(dirPath)
+	{
+		var dirPathItem;
+
+		switch (gettype(dirPath)) {
+			case "string":
+				this->_registerDir(dirPath);
+				break;
+			case "array":
+			case "object":
+				for dirPathItem in iterator(dirPath) {
+					this->_registerDir(dirPathItem);
+				}
+				break;
+			default:
+				throw new Exception("Invalid services.");
```

Comment:
I would like to see here a error message with an explanation of what happened and the reasons for this